lib/fog/dynect/models/dns/record.rb

```diff
@@ -59,6 +51,25 @@ def zone
         def zone=(new_zone)
           @zone = new_zone
         end
+
+        def refresh_id
+          begin 
+            records = service.get_record(type, zone.identity, name).body['data']
+          rescue
+            return
+          end
+          # data in format ['/REST/xRecord/domain/fqdn/identity]
+          records.map! do |record|
+            tokens = record.split('/')
+            {
+              :identity => tokens.last,
+              :type     => tokens[2][0...-6] # everything before 'Record'
+            }
+          end
+          record = records.find(ifnone={}) {|record| record[:type] == type}
```

Comment:
From http://ruby-doc.org/core-1.9.3/Enumerable.html#method-i-find
> If no object matches, calls ifnone and returns its result when it is specified...

It's been a while, but I believe this was an attempt to prevent passing a `nil` to `merge_attributes`.  However, the value passed to ifnone would need to be callable for that to work; something like: `records.find(ifnone=Proc.new { {} })`

That code may be left over from an intermediate attempt at fixing this, before I understood the real problem.  I can test removing that and make sure it all still works.